tools/socktap/CMakeLists.txt

```diff
@@ -27,7 +27,23 @@ add_executable(socktap
     udp_link.cpp
 )
 
-target_link_libraries(socktap PUBLIC Boost::system Boost::program_options Threads::Threads vanetza)
+option(SOCKTAP_WITH_AUTOTALKS "Use Autotalks API for socktap" OFF) # Both Secton and Craton devices
+if (SOCKTAP_WITH_AUTOTALKS)
+    find_package(Autotalks MODULE REQUIRED)
+    target_compile_definitions(socktap PUBLIC "SOCKTAP_WITH_AUTOTALKS")
+    if(NOT AUTOTALKS_CRATON)
+        target_compile_definitions(socktap PUBLIC "SECTON")
+        target_link_libraries(socktap PUBLIC Boost::system Boost::program_options Threads::Threads vanetza Autotalks::AtlkRemote Autotalks::ToMcrypt Autotalks::ToMmath ${AUTOTALKS_EXECUTABLES} rt)
```

Comment:
Looks like *Autotalks::AtlkRemote* depends on *ToMcrypt* and *ToMath*? You could specify this dependency in *FindAutotalks.cmake* and also add the *AUTOTALKS_EXECUTABLES* (objects) to the *AtlkRemote* target.
Ideally, you just state *Autotalks::AtlkRemote* as dependency here and everything else (including rt) is then transitively added by CMake.

Reply:
Turns out _AtlkRemote_ does not even need _ToMcrypt_ and _ToMmath_. However, I moved the dependencies to _FindAutotalks.cmake_ as you suggested, I thinks now it is correct.